apollo-router/src/plugins/connectors/plugin.rs

```diff
@@ -143,6 +148,49 @@ impl Plugin for Connectors {
             )
             .boxed()
     }
+
+    fn execution_service(&self, service: execution::BoxService) -> execution::BoxService {
+        ServiceBuilder::new()
+            .map_request(|req: execution::Request| {
+                let Some(connectors) = get_connectors(&req.context) else {
+                    return req;
+                };
+
+                // add [{"subgraph_name": "", "source_name": ""}] to the context
+                // for connectors with sources in the query plan.
+                let list = req
+                    .query_plan
+                    .root
+                    .service_usage()
+                    .unique()
+                    .flat_map(|service_name| {
+                        let Some(connector) = connectors.get(service_name) else {
+                            return None;
+                        };
+
+                        let Some(ref source_name) = connector.id.source_name else {
+                            return None;
+                        };
+
+                        Some((connector.id.subgraph_name.clone(), source_name.clone()))
+                    })
+                    .unique()
+                    .map(|(subgraph_name, source_name)| {
+                        json!({
+                            "subgraph_name": subgraph_name,
+                            "source_name": source_name,
+                        })
+                    })
```

Comment:
I previously added a `ConnectorSourceRef` type that is exactly this tuple. This should let you simplify this to:

```rust
                   .flat_map(|service_name| {
                        connectors
                            .get(service_name)
                            .map(|connector| ConnectorSourceRef::try_from(connector).ok())
                    })
                   .unique()
```

You'd need to derive `Serialize` and `Deserialize` for `ConnectorSourceRef`, and add something like the following:

```rust
impl TryFrom<&Connector> for ConnectorSourceRef {
    type Error = ();

    fn try_from(value: &Connector) -> Result<Self, Self::Error> {
        Ok(Self {
            subgraph_name: value.id.subgraph_name.to_string(),
            source_name: value.id.source_name.clone().ok_or(())?,
        })
    }
}
```